src/caf/space/inputs.py

```diff
@@ -157,6 +161,59 @@ def _valid_data_col(cls, v, values):
         return v
 
 
+@dataclasses.dataclass
+class SpaceArguments:
+    """Command Line arguments for running space."""
+
+    config_path: Path
+    mode: str
+    out_path: Path
+
+    @classmethod
+    def parse(cls) -> SpaceArguments:
+        """Parse command line argument."""
+        parser = argparse.ArgumentParser(
+            description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+        parser.add_argument(
```

Comment:
I don't think this is necessary here but for future reference you can use `add_subparser` to have separate sub-commands which have their own set of arguments (and some shared arguments), see [argparse sub-commands](https://docs.python.org/3/library/argparse.html#sub-commands).

For example you could have sub-command "GUI" with no arguments and sub-commands sub-commands "spatial" and "weighted" which have mandatory config path and output folder arguments.